crates/cxx-qt-lib-headers/include/qml/qqmlapplicationengine.h

```diff
@@ -11,13 +11,17 @@
 #include <memory>
 
 #include <QtQml/QQmlApplicationEngine>
+#include <QtQml/QQmlEngine>
 
 namespace rust {
 namespace cxxqtlib1 {
 
 ::std::unique_ptr<QQmlApplicationEngine>
 qqmlapplicationengineNew();
 
+QQmlEngine&
+qqmlapplicationengineAsQQmlEngine(QQmlApplicationEngine&);
```

Comment:
This sounds like a good candidate for another template in common.h:
```cpp
template<typename Base, typename T>
Base & upcast(T& t)
{
   // Maybe add some static_assert that Base is indeed base class of T
    return static_cast<Base&>(t);
}
```

Reply:
yup, lets add it to #823